ncdfchecker.py

```diff
@@ -280,7 +334,22 @@ def simple_variable_checks(product, constraints, strict=False, logger=None):
                             if intervalkey in product.variables:
                                 arr = product[intervalkey][:]
                                 step = constraints[variable][key][intervalkey]
-                                if not check_stepsize(arr, step):
+
+                                monthly = step == 'mon'
+                                if monthly:
+                                    logger.info('This is a monthly file.')
+                                    if initdate is None or runlength is None:
```

Comment:
This doesn't make sense for checking if dates are monthly, it only makes sense for a validity date range - again though, as mentioned elsewhere I'm not sure that's necessary here.